controllers/platformoperator_controller.go

```diff
@@ -62,47 +67,82 @@ func (r *PlatformOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	log.Info("reconciling request", "req", req.NamespacedName)
 	defer log.Info("finished reconciling request", "req", req.NamespacedName)
 
-	// TODO: flesh out status condition management
 	po := &platformv1alpha1.PlatformOperator{}
 	if err := r.Get(ctx, req.NamespacedName, po); err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 	defer func() {
 		po := po.DeepCopy()
 		po.ObjectMeta.ManagedFields = nil
+
 		if err := r.Status().Patch(ctx, po, client.Apply, client.FieldOwner("platformoperator")); err != nil {
 			log.Error(err, "failed to patch status")
 		}
 	}()
 
-	desiredBundle, err := r.Sourcer.Source(ctx, po)
+	return r.reconcile(ctx, po)
+}
+
+func (r *PlatformOperatorReconciler) reconcile(ctx context.Context, po *platformv1alpha1.PlatformOperator) (ctrl.Result, error) {
+	bd, err := r.ensureDesiredBundleDeployment(ctx, po)
 	if err != nil {
+		reason := platformtypes.ReasonApplyFailed
+		status := metav1.ConditionUnknown
```

Comment:
Should status just always be `"False"`? Both reasons seem to indicate that the BD has not been applied.

Reply:
Updated in https://github.com/openshift/platform-operators/pull/41/commits/28b4316bc8c76cdb4d61f41877fbe6351ac50f39.